src/css-token-stream-value.js

```diff
@@ -0,0 +1,32 @@
+(function(internal, scope) {
+
+    function CSSTokenStreamValue(values) {
+        if (values == undefined) {
+            values = [];
+        }
+        if (!Array.isArray(values)) {
+            throw new TypeError('CSSTokenStreamValue should be an array of string');
+        }
+        for (var i = 0; i < values.length; i++) {
+            if (typeof values[i] != 'string') {
+                throw new TypeError("CSSTokenStreamValue's elements should be strings");
+            }
+        }
+        this.values = values;
+        // TODO: may need to be changed like 
+        //       https://drafts.css-houdini.org/css-typed-om/#tokenstreamvalue-normalization
+    }
+
+    internal.inherit(CSSTokenStreamValue, CSSStyleValue);
+
+    CSSTokenStreamValue.prototype.size = function() {
+        return this.values.length;
+    }
+
+    CSSTokenStreamValue.prototype.referenceAtIndex = function(index) {
```

Comment:
Wasn't this method just an implementation detail in the C++ version? You can implement iteration in JS using https://developer.mozilla.org/en-US/docs/Web/JavaScript/Guide/Iterators_and_Generators